### DISCUSSION TITLE
add an event handler to an object doesn't work

### DISCUSSION THREAD

**Discussion by olivia1520** [fabricjs/fabric.js#9926] [Q&A]
When I click the rect, the console only prints "click the canvas" instead of "click the rect". Is there any solution? Thanks.
`var rect = new fabric.Rect({
    left: 100,
    top: 100,
    fill: 'red',
    width: 100,
    height: 100
  });

canvas.add(rect);

rect.on('mouse:down', function(options){
    console.log("click the rect");
    })

canvas.on('mouse:down',function(options){
    console.log("click the canvas");
})`

**Comment by asturur** [comment]
rect has 'mousedown' without the double colon. i know is confusing but TS should help now.
